application/src/main/java/org/togetherjava/tjbot/utils/ModAuditLogWriter.java

```diff
@@ -0,0 +1,61 @@
+package org.togetherjava.tjbot.utils;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.interaction.GenericInteractionCreateEvent;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class ModAuditLogWriter {
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static final Predicate<String> isAuditLogChannelName =
+            Pattern.compile(Config.getInstance().getModAuditLogChannelPattern()).asMatchPredicate();
+    private static final Predicate<TextChannel> isAuditLogChannel =
+            channel -> isAuditLogChannelName.test(channel.getName());
+
+    private ModAuditLogWriter() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    /**
+     * logs an entry in the mod audit log channel.
+     * 
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param embed the embed that will be sent.
+     *
+     * @param files the files added to the message.
+     */
+    public static synchronized void log(@NotNull Guild guild, @NotNull EmbedBuilder embed,
+            VirtualFile... files) {
+        Optional<TextChannel> auditLogChannel = getModAuditLogChannel(guild);
+        if (auditLogChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to log moderation events, did not find a mod audit log channel matching the configured pattern '{}' for guild '{}'",
+                    Config.getInstance().getModAuditLogChannelPattern(), guild.getName());
+            return;
+        }
```

Comment:
I can't think of a way to solve it, because we have to use the `Optional` to make sure that the channel exists. I think that moving the warning to the method (after merging the duplicate methods) is the best solution, but I think we have to keep the part that checks if its empty where it is.

Reply:
Well, after a second look, I think the part that bummers me is that you have the `Config.getInstance().getModAuditLogChannelPattern()` in the log. This assumes the implementation of `getModAuditLogChannel`, which is not good and can easily get out of sync.

I appreciate that you wanted to give extra details in the log but it is probably better to remove that part from the log all together. And then you can of course also keep it here.

Reply:
in fact, I've copied this part from `ModAuditLogRoutine`. I'll merge the duplicate 2 `getModAuditLogChannel()` methods,
and I think it'll be best to move the debug log into the method. so then we'll only have
```java
Optional<TextChannel> auditLogChannel = getModAuditLogChannel(guild);
if (auditLogChannel.isEmpty()) {
    return;
}
```
outside of it

Reply:
done.